File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, nums: List[int]) -> int:
        if reversed(nums) ==  nums : return 0
        prev_low = 0
        max_profit = 0
        for i in range(0,len(nums)):
            if(nums[i]<nums[prev_low]):
                prev_low = i
            max_profit = max(max_profit, nums[i]-nums[prev_low])
        return max_profit

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0, buy=prices[0], sell, size = prices.size();
        for(int i=0; i<size; i++){
            buy = min(buy, prices[i]);
            profit = max(profit, prices[i] - buy);
        }
        return profit;
    }
};